.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,88 @@
+name: Pull Request - Java Build & Lint & beta release
+
+on: [pull_request]
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: '17.0.1'
+
+    - name: Run Linter
+      run: bash .githooks/pre-commit.sh
+
+    - name: Cache Gradle dependendies
+      uses: actions/cache@v1
+      with:
+        path: ~/.gradle/caches
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+ 
+    - name: Build with Gradle
+      run: bash ./gradlew build --info
+
+    - name: Load Release JSON Config
+      uses: rgarcia-phi/json-to-variables@v1.1.0
+      with:
+        filename: 'release-config.json'
+        prefix: release
+
+    - name: Show Release Version
+      id: release-info
+      run: |
+        git fetch --tags
+        last_version=$(git tag -l "${{ env.release_major_version }}.${{ env.release_minor_version }}.*" --sort=v:refname | tail -1)
+        last_patch=$(cut -d '.' -f 3 <<< "$last_version")
+        new_patch="$(($last_patch + 1))" # TODO: Start from zero if not set
+        my_version="${{ env.release_major_version }}.${{ env.release_minor_version }}.$new_patch"
+        echo "Output [$my_version]"
+        echo "version=$my_version" >> $GITHUB_OUTPUT 
+
+    - name: Create Release
+      id: release
+      uses: avakar/tag-and-release@v1
+      with:
+        tag_name: "${{ steps.release-info.outputs.version }}"
+        release_name: "BETA ${{ steps.release-info.outputs.version }}"
```

Comment:
Want to find a way to update the name and description based off the MR or commits